src/docs/product/performance/queries.mdx

```diff
@@ -84,17 +104,36 @@ You can see the full query by hovering on a truncated description, or clicking i
 
 To open a query's **Query Summary** page, click on the query from either the table in the "Most Time-Consuming Queries" widget on the **Performance** page or the table in the **Queries** page.
 
-At the top of the page, queries per minute, average duration, and time spent are shown for the selected time range. The full query is shown below these metrics, followed by graphs for queries per minute and average duration.
+At the top of the page, queries per minute, average duration, and time spent are shown for the selected time range. The full query is shown below these metrics, followed by graphs for queries per minute and average duration. The query's source in the code is also shown if it is available.
 
 At the bottom, you can find a list of endpoints the query is found in, sorted by the most time your application spent in that span.
 
 If you want to investigate a specific endpoint, click on it to open a sidebar showing some sample events.
 
 The gif below provides a detailed walkthrough of the **Query Summary** page.
 
-<div style="position: relative; padding-bottom: calc(48.68055555555556% + 41px); height: 0; width: 100%"><iframe src="https://demo.arcade.software/MTzviZtIiN7ZCmNFVJg4?embed" frameborder="0" loading="lazy" webkitallowfullscreen mozallowfullscreen allowfullscreen style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;color-scheme: light;" title="Performance - Queries - Query Summary Page Walkthrough"></iframe></div>
+<div style="position: relative; padding-bottom: calc(48.68055555555556% + 41px); height: 0; width: 100%">
+  <iframe
+    src="https://demo.arcade.software/MTzviZtIiN7ZCmNFVJg4?embed"
+    frameborder="0"
+    loading="lazy"
+    webkitallowfullscreen
+    mozallowfullscreen
+    allowfullscreen
+    style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;color-scheme: light;"
+    title="Performance - Queries - Query Summary Page Walkthrough"
+  ></iframe>
+</div>
+
+## Query Sources
+
+Query Sources are currently supported in Sentry's Laravel and Django SDKs.
 
-</br>
+A query's source is the location in your application's code that triggered the query. Sentry automatically determines the query source by examining the stack trace and attaches the source to the query data. To reduce the performance impact, Sentry only attaches query sources to queries that are slower than a <PlatformLink to="/configuration/options">configurable threshold</PlatformLink>. If you want to see more query sources, you can reduce this threshold.
```

Comment:
Can we specify which config option users should configure?

I found this [python config option](https://docs.sentry.io/platforms/python/configuration/options/#enable-db-query-source) (which I assume works for Django) says query source is disabled by default. So would users need to enable it first? If so, we should call this out in line 130 as a pre-req. And I assume the threshold config is `db_query_source_threshold_ms`.

For [laravel](https://docs.sentry.io/platforms/php/guides/laravel/configuration/laravel-options/), do users also need to enable this config option`'sql_origin' => env('SENTRY_TRACE_SQL_ORIGIN_ENABLED', true),`? Not sure how to configure the threshold though, so might need more instruction.

Since this only works for 2 SDKs atm, I think we should be more explicit with the instructions here. Tbh, I had trouble finding the config options, so I'd expect it to be frustrating for users as well.

Reply:
Oof, good questions. This is a little complicated!

- the supported SDKs are Laravel and Python (not just Django). Laravel is a single entire platform that supports this feature, in that framework. In the Python platform, several integrations are supported (Django, SQLAlchemy, asyncpg)
- right now, in Laravel the feature is enabled by default. The `SENTRY_TRACE_SQL_ORIGIN_ENABLED` option controls it, like you said. Also, there's no way to control the slowness threshold. This is because this is a pre-existing feature. Python and all future platforms will have the threshold
- in Python, the feature is currently disabled by default, but it will be enabled by default in a more recent SDK version (available before this feature is released)

What's the best way to handle all this weird complexity?